site/content/integrate/apps/functionality/external-webhooks/_index.md

```diff
@@ -7,87 +7,194 @@ aliases:
+
+Using the example from the previous section, the webhook URL will look like this:
+
+`http://my-mattermost-server/plugins/com.mattermost.apps/apps/my-app/webhook/my-sub-path?secret=cwsjhrdqebyf8qrnabtxio7k7r`
+
+The Mattermost server will authenticate incoming webhook requests by comparing the `secret` URL query value with the App's `webhook_secret`.
+If an incoming webhook request does not include the correct secret, the request will be denied with an HTTP 401 response.
+
+{{<note "Disable authentication:">}}
+It is possible to disable webhook authentication by setting the `remote_webhook_auth_type` App manifest field to `none`.
+The Mattermost server will not generate a webhook secret, will not perform any authentication against the call request, and will pass all webhook calls to the appropriate App call handlers.
+This may have security and hosting cost implications.
+{{</note>}}
+
+### Get the webhook secret
+
+A simple way to get the App's webhook secret is to use a slash command that executes a [call]({{<ref "/integrate/apps/structure/call" >}}) with the `app` expand field set to `all`.
+
+For example, the following bindings create a slash command that will provide the webhook secret to its call:
+
+```json
+[
+    {
+        "location": "/command",
+        "bindings": [
+            {
+                "location": "configure",
+                "label": "configure",
+                "description": "Configure the app",
+                "submit": {
+                    "path": "/configure",
+                    "expand": {
+                        "app": "all"
+                    }
+                }
+            }
+        ]
+    }
+]
+```
+
+The call request would look like this:
 
 ```json
 {
-    "path": "/installed",
+    "path": "/configure",
+    "expand": {
+        "app": "all"
+    },
     "context": {
-        "app_id": "my-app",
+        "app_id": "hello-world",
+        "location": "/command/configure",
+        "user_agent": "webapp",
+        "track_as_submit": true,
+        "mattermost_site_url": "http://mattermost:8066",
         "developer_mode": true,
+        "app_path": "/plugins/com.mattermost.apps/apps/hello-world",
+        "bot_user_id": "mgbd1czngjbbdx6eqruqabdeie",
+        "bot_access_token": "q8idzs7dspf8ugra9sb7dkgxwe",
         "app": {
-            "app_id": "my-app",
+            "app_id": "hello-world",
             "version": "0.1.0",
-            "webhook_secret": "9a44ckeqytd3bftn3c3y53968o",
-            "bot_user_id": "7q7kaakokfdsdycy3pr9ctkc5r",
-            "bot_username": "my-app"
+            "webhook_secret": "cwsjhrdqebyf8qrnabtxio7k7r",
+            "bot_user_id": "mgbd1czngjbbdx6eqruqabdeie",
+            "bot_username": "hello-world",
+            "remote_oauth2": {}
+        },
+        "acting_user": {
+            "id": "7q7kaakokfdsdycy3pr9ctkc5r"
+            // additional fields omitted for brevity
```

Comment:
If we set this json block to be `js` instead of `json`, this comment won't be highlighted in red here. The syntax highlighting may be a bit different though. Maybe we can omit this comment since it's not integral to the example?

Reply:
I would like to keep this here. I've used it in many places to indicate that there are more fields the implementer can expect in the output.